base58.py

```diff
@@ -23,108 +24,168 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-"""encode/decode base58 in the same way that Bitcoin does"""
+# This module is based upon base58 snippets found scattered over many bitcoin
+# tools written in python. From what I gather the original source is from a
+# forum post by Gavin Andresen, so direct your praise to him.
+# This module adds shiny packaging and support for python3.
+
+from hashlib import sha256
+
+__version__ = '1.0.3'
+
+# 58 character alphabet used
+alphabet = b'123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
+
+
+if bytes == str:  # python2
```

Comment:
can we use the sys package instead to check the python version?  https://stackoverflow.com/questions/9079036/how-do-i-detect-the-python-version-at-runtime

edit: if this is copied from elsewhere we should have a clear link to it, preferably in the file itself.

Reply:
The source of this code is explained in the commit message for 1a806ba5c4fb104f8e8ba5c9b43d4d621d7ec477, and 62def076239f0b3a616cf84067f61b13d108fd1d, and in commit 1181e756f5a99bc65ab5d19ce07b11bb0b2c7199 the link was added to the glacierscript.py source.